src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectStrategy.java

```diff
@@ -110,23 +116,95 @@ public AuthorizeProjectStrategyDescriptor getDescriptor() {
                     Jenkins.getAuthentication().getName()
             ));
         }
     }
     
     /**
      * Tests if the job can be reconfigured by the current user when this strategy is the configured strategy.
+     * Users with {@link Jenkins#ADMINISTER} permission skips this check.
      *
      * @param context the context of the job
      * @return {@code true} if and only if the current user is allowed to reconfigure the specified job.
      * @since 1.3.0
      */
-    public boolean hasConfigurePermission(AccessControlled context) {
+    public boolean hasJobConfigurePermission(AccessControlled context) {
         return true;
     }
 
+    /**
+     * Checks that the authorization can be configured by the current user.
+     *
+     * @param context the context of the job
+     * @throws AccessDeniedException if the current user is not allowed to configure this authorization
+     * @since 1.3.0
+     */
+    public final void checkAuthorizationConfigurePermission(AccessControlled context) {
+        if (!hasAuthorizationConfigurePermission(context)) {
+            throw new AccessDeniedException(Messages.AuthorizeProjectStrategy_UserNotAuthorized(
+                    Jenkins.getAuthentication().getName()
+            ));
+        }
+    }
+
+    /**
+     * Tests if the authorization can be configured by the current user.
+     * Users with {@link Jenkins#ADMINISTER} permission skips this check.
+     *
+     * @param context the context of the job
+     * @return {@code true} if and only if the current user is allowed to configure this authorization.
+     * @since 1.3.0
+     */
+    public boolean hasAuthorizationConfigurePermission(AccessControlled context) {
+        return true;
+    }
+
+    /**
+     * If we are being deserialized outside of loading the initial jobs (or reloading) then we need to cross check
+     * the strategy permissions to defend against somebody trying to push a configuration relating to a user other
+     * than themselves.
+     *
+     * @return {@code this}
+     * @throws ObjectStreamException if the object cannot be deserialized.
+     * @see AuthorizeProjectProperty#setStrategyCritical()
+     */
+    protected Object readResolve() throws ObjectStreamException {
+        checkUnsecuredConfiguration();
+        return this;
+    }
+
+    private void checkUnsecuredConfiguration() throws ObjectStreamException {
+        Authentication authentication = Jenkins.getAuthentication();
+        if (authentication == ACL.SYSTEM) {
+            // It is considered to initial loading or reloading.
+            return;
+        }
+        if (!ProjectQueueItemAuthenticator.isConfigured()) {
+            return;
+        }
+        if (Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER)) {
+            // allows any configurations by system administrators.
+            // It may not be allowed even if the user is an administrator of the job.
+            return;
+        }

```

Comment:
I found that testing `ADMINISTER` against `context` is dangerous, as malicious users may create a job with ADMINISTER permission (even though I'm not sure one can do that).
Test `ADMINISTER` permissions should be performed against `Jenkins.getActiveInstance()`.